### PR TITLE
[dcl.enum] Merge duplicate normative paragraphs on redeclarations.

### LINKED ISSUE #1826: [dcl.enum] Duplicate sentences about redeclarations
[\[dcl.enum\]/4](http://eel.is/c++draft/dcl.enum#4.sentence-1):

> If the enum-key is followed by a nested-name-specifier, the enum-specifier shall refer to an enumeration that was previously declared directly in the class or namespace to which the nested-name-specifier refers (i.e., neither inherited nor introduced by a using-declaration), and the enum-specifier shall appear in a namespace enclosing the previous declaration.

Is mostly repeated in [\[dcl.enum\]/12](http://eel.is/c++draft/dcl.enum#12.sentence-1):

> If an enum-head contains a nested-name-specifier, the enum-specifier shall refer to an enumeration that was previously declared directly in the class or namespace to which the nested-name-specifier refers, or in an element of the inline namespace set ([namespace.def]) of that namespace (i.e., not merely inherited or introduced by a using-declaration), and the enum-specifier shall appear in a namespace enclosing the previous declaration.

One of these was probably intended to replace the other. I don't know which.

### PR DESCRIPTION
**Pull request by jensmaurer**
Fixes #1826.

### CODE REVIEW

File: source/declarations.tex [hunk 1]
```diff
@@ -2040,10 +2040,15 @@
 If the \grammarterm{enum-key} is followed by a
```
Comment (zygoloid) [inline source/declarations.tex]: Please add "in an enum-head or opaque-enum-declaration" after "enum-key". I think we may have duplicated this previously as a misguided and unclear attempt to address both cases separately.

Comment (zygoloid) [reply source/declarations.tex]: Hmm, could we refactor opaque-enum-declaration to use enum-head-name, and reference that here? That would avoid confusion over whether this applies if there is an intervening attribute-specifier-seq.

**Comment by tkoeppe** [comment]
@jensmaurer: actionable comments here

**Comment by jensmaurer** [comment]
Actions performed. @zygoloid is next.

**Comment by zygoloid** [comment]
Looks good, let's apply this after the Jacksonville motions.

File: source/declarations.tex [hunk 2]
```diff
@@ -2038,12 +2038,18 @@
 underlying type as in the original declaration.
 
 \pnum
-If the \grammarterm{enum-key} is followed by a
-\grammarterm{nested-name-specifier}, the \grammarterm{enum-specifier} shall
+If an \grammarterm{enum-head-name} contains a
+\grammarterm{nested-name-specifier}, the latter
+shall not begin with a \grammarterm{decltype-specifier} and
```
Comment (zygoloid) [inline source/declarations.tex]: Was this `decltype` rule for *opaque-enum-declaration*s present before (maybe implied by pre-existing wording)? I see we had such a rule for *enum-head*s.

Comment (jensmaurer) [reply source/declarations.tex]: Well, we have (unchanged from before):

If an \grammarterm{opaque-enum-declaration} contains
a \grammarterm{nested-name-specifier},
the declaration shall be an explicit specialization\iref{temp.expl.spec}.

and we have 

11.3 [dcl.meaning]
The nested-name-specifier of a qualified declarator-id shall not begin with a decltype-specifier.

That seems close enough to me. (it seems we don't want redeclarations to use decltype to nominate the entity being redeclared. Which is totally reasonable.)

**Comment by jensmaurer** [comment]
Rebased.

File: source/declarations.tex (hunk 2 above)
Comment (zygoloid) [reply source/declarations.tex]: I agree that we don't want to allow `decltype` here, and indeed current implementations don't distinguish between the opaque enum declaration case and the regular enum case, but if this is a normative change (as it appears to be), we should fix that portion of this via the CWG issue process.

Comment (jensmaurer) [reply source/declarations.tex]: Fixed by limited the prohibition to the enum-head case.

**Comment by tkoeppe** [comment]
@zygoloid says:

> I think what I'm most worried about here is that making the editorial change first turns the wording from "wrong but somewhat ambiguous" (it looks like we just forgot the rule) to "unambiguously wrong" (the difference between the two cases is clearly intentional), which I would expect to make it more likely for a user or implementer to get confused or do the wrong thing.

Let's run this by CWG then.

**Comment by jensmaurer** [comment]
Rebased.

**Comment by jensmaurer** [comment]
CWG in San Diego: Fine with the change.

http://wiki.edg.com/pub/Wg21sandiego2018/CoreWorkingGroup/cwg-ed1827.html

**Pull request merged**